packages/context/src/binding.ts

```diff
@@ -328,11 +349,26 @@ export class Binding<T = BoundValue> {
    */
   applyDefaultScope(scope: BindingScope): this {
     if (!this._scope) {
-      this._scope = scope;
+      this.inScope(scope);
     }
     return this;
   }
 
+  /**
+   * Set the `_getValue` function
+   * @param getValueFn getValue function
+   */
+  private _setValueGetter(
+    getValueFn: (
+      ctx: Context,
+      optionsOrSession?: ResolutionOptionsOrSession,
+    ) => ValueOrPromise<T | undefined>,
```

Comment:
Have you considered defining a named type, e.g. `BindingValueGetter`? This type definition is repeated in two places (the type of `_getValue` property, the type of `_setValueGetter` argument).